### PR TITLE
rime-cli: init at 0.0.3

### LINKED ISSUE #214147: rime-cli
**Project description**
A tool for rime IME to add vocabulary, which is a dependency of [coc-rime](https://github.com/tonyfettes/coc-rime).

Now
```sh
patchelf --set-interpreter "$(nix-build --no-out-link '<nixpkgs>' -A glibc)"/lib/ld-linux-*.so.* rime-cli
patchelf --set-rpath "$(nix-build --no-out-link '<nixpkgs>' -A glibc)/lib:$(nix-build --no-out-link '<nixpkgs>' -A json_c)/lib:$(nix-build --no-out-link '<nixpkgs>' -A librime)/lib" rime-cli
```

can make the program work.

**Metadata**

* homepage URL: <https://github.com/puddinging/rime-cli>
* source URL: <https://github.com/puddinging/rime-cli>
* license: apache
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #214147

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by figsoda** [comment]
@freed-wu the rime-cli you [linked](https://github.com/puddinging/rime-cli) to seems to be different from the [one](https://github.com/tonyfettes/rime-cli) [coc-rime](https://github.com/tonyfettes/coc-rime) [depends](https://github.com/tonyfettes/coc-rime/blob/master/README.zh_cn.md#%E5%86%99%E5%9C%A8%E5%89%8D%E9%9D%A2) on

File: pkgs/applications/office/rime-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    rev = "v${version}";
+    hash = "sha256-CI0Jva7oA/zUGatv+wCdByqbTBNQRw+4clr8IDKX6HQ=";
+  };
+
+  vendorHash = null;
+
+  meta = with lib; {
+    homepage = "https://github.com/puddinging/rime-cli";
+    changelog = "https://github.com/puddinging/rime-cli/releases/tag/v${version}";
+    description = "关于小狼毫输入法的一个小工具";
```
Comment (bbigras) [inline pkgs/applications/office/rime-cli/default.nix:19]: I don't know if there's a rule for that but shouldn't the description be in English?

Comment (urandom2) [reply pkgs/applications/office/rime-cli/default.nix:19]: that is what their github repo lists as the cli description, and I was hesitant to trust google translate:
```suggestion
    description = "A small tool about Xiaolanghao input method";
```

Comment (SharzyL) [reply pkgs/applications/office/rime-cli/default.nix:19]: I suggest describing it as “A command line tool to add customized vocabulary for Rime IME”

**Comment by Freed-Wu** [comment]
> @Freed-Wu the rime-cli you linked to seems to be different from the one coc-rime depends on

It's my fault. [coc-rime](https://github.com/tonyfettes/coc-rime) use this [rime-cli](https://github.com/tonyfettes/rime-cli). I use it by a flake.nix. In that flake.nix, I name it as rime-cli-server to avoid conflict. Maybe we can provide packages in nixpkgs for coc-rime and rime-cli-server? @tonyfettes, maybe we can rename [rime-cli](https://github.com/tonyfettes/rime-cli) to rime-cli-server to avoid conflict?

**Comment by SuperSandro2000** [comment]
You linked two times the same rime-cli

**Pull request merged**